### PR TITLE
39 Create Agent Module 

### LINKED ISSUE #39: Create Agent module
## Background
Atum agent creation that will be included in the apps/libs using Atum

## Feature
Create a module for atum agent within the repository. This means converting the Atum project into a multi-modile one. 

## Proposed Solution
Solution Ideas:
1. It's probably enough to support Scala 2.12 and Spark 3.x

### PR DESCRIPTION
**Pull request by seldon851**
The ticket --> https://github.com/AbsaOSS/atum-service/issues/39

Closes #39

### CODE REVIEW

File: agent/src/main/scala/za/co/absa/atum/agent/Hello.scala [hunk 1]
```diff
@@ -0,0 +1,10 @@
+
+package za.co.absa.atum.agent
+
+object Hello extends App {
+
+  println("running")
```
Comment (benedeki) [inline agent/src/main/scala/za/co/absa/atum/agent/Hello.scala:6]: :-D

File: scalastyle-config.xml [hunk 2]
```diff
@@ -0,0 +1,161 @@
+<!--
+  ~ Copyright 2018 ABSA Group Limited
```
Comment (benedeki) [inline scalastyle-config.xml:2]: Probably taken from some other project (which one? 😉 )
The license year should be the year of project establishment. (2021 this one, I believe)

File: random-dataset/README.md [hunk 3]
```diff
@@ -0,0 +1 @@
+This folder contains random data for testing
```
Comment (benedeki) [inline random-dataset/README.md:1]: I would keep this folder for future PRs, probably in `test/resources/data` directory.

File: project/Dependencies.scala [hunk 4]
```diff
@@ -16,36 +16,177 @@
...
+      springFoxSwaggerUI,
+      springFoxBoot,
+      scalaJava8Compat,
+      jacksonModuleScala
+    )
+  }
+
+}
+
+object AgentDependencies {
```
Comment (benedeki) [inline project/Dependencies.scala:59]: Why so many dependencies?
I would actually keep this empty at start. And add items there as needed when actual code arrives into the agent.

File: agent/src/main/scala/za/co/absa/atum/agent/com/dummy/Person.scala [hunk 5]
```diff
@@ -0,0 +1,17 @@
+
+package za.co.absa.atum.agent.com.dummy
+
+case class Person(
```
Comment (benedeki) [inline agent/src/main/scala/za/co/absa/atum/agent/com/dummy/Person.scala:4]: I think the above is enough to have some dummy code. But no problem if you keep this one too.

File: scalastyle-config.xml (hunk 2 above)
Comment (seldon851) [reply scalastyle-config.xml:2]: From Atum

File: project/Dependencies.scala (hunk 4 above)
Comment (seldon851) [reply project/Dependencies.scala:59]: agree

File: random-dataset/README.md (hunk 3 above)
Comment (seldon851) [reply random-dataset/README.md:1]: perfect

File: scalastyle-config.xml (hunk 2 above)
Comment (seldon851) [reply scalastyle-config.xml:2]: changed

File: project/Dependencies.scala (hunk 4 above)
Comment (benedeki) [reply project/Dependencies.scala:59]: I still seem them here. 
For example I don't think this will ever need AWS dependency. 🤔

File: build.sbt [hunk 6]
```diff
@@ -46,9 +46,9 @@ lazy val server = project
 lazy val agent = project
   .settings(
   name := "atum-agent",
-    libraryDependencies ++= (
-      AgentDependencies.rootDependencies(scalaVersion.value) ++
-        AgentDependencies.coreDependencies(scalaVersion.value)),
+    /*libraryDependencies ++= (
```
Comment (benedeki) [inline build.sbt:49]: No dead code in the master branch, unless it's a example.

**Review by benedeki — APPROVED** [review]
- [x] code reviewed
- [x] pulled
- [ ] built
- [ ] run

**Comment by seldon851** [comment]
thank you ,@benedeki

**Pull request merged**